library/src/app/components/bank-account-management/bank-account-connect/bank-account-connect.component.spec.ts

```diff
@@ -0,0 +1,223 @@
+        { provide: ConfigService, useValue: MockConfigService },
+        { provide: RoutingService, useValue: MockRoutingService },
+        { provide: BankAccountService, useValue: MockBankAccountService },
+        { provide: BanksService, useValue: MockBankService },
+        Renderer2
+      ],
+      schemas: [CUSTOM_ELEMENTS_SCHEMA]
+    }).compileComponents();
+    MockEventService = TestBed.inject(EventService);
+    MockErrorService = TestBed.inject(ErrorService);
+    MockConfigService = TestBed.inject(ConfigService);
+    MockRoutingService = TestBed.inject(RoutingService);
+    MockBankService = TestBed.inject(BanksService);
+    MockBankService.getBank.and.returnValue(of(TestConstants.BANK_BANK_MODEL));
+    MockBankAccountService = TestBed.inject(BankAccountService);
+    MockBankAccountService.createWorkflow.and.returnValue(
+      of(TestConstants.WORKFLOW_BANK_MODEL)
+    );
+    MockBankAccountService.getWorkflow.and.returnValue(
+      of(TestConstants.WORKFLOW_BANK_MODEL_WITH_DETAILS)
+    );
+    MockBankAccountService.getPlaidClient.and.returnValue(of(false));
+
+    // Reset call count
+    MockBankAccountService.getPlaidClient.calls.reset();
+
+    fixture = TestBed.createComponent(BankAccountConnectComponent);
+    component = fixture.componentInstance;
+    component.plaidScriptSrc = '/assets/plaidTestScript.js';
+
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    component.bootstrapPlaid = () => undefined;
+    expect(component).toBeTruthy();
+  });
+
+  it('should add account', () => {
+    component.bootstrapPlaid = () => undefined;
+    component.onAddAccount();
+
+    expect(MockConfigService.getConfig$).toHaveBeenCalled();
+    expect(MockBankAccountService.createWorkflow).toHaveBeenCalled();
+  });
+
+  it('should handle errors on addAccount', () => {
+    component.bootstrapPlaid = () => undefined;
+    MockBankAccountService.createWorkflow.and.returnValue(error$);
+
+    component.onAddAccount();
+
+    expect(MockErrorService.handleError).toHaveBeenCalled();
+    expect(MockEventService.handleEvent).toHaveBeenCalled();
+  });
+
+  it('should bootstrap the Plaid sdk', () => {
+    // Reset call count
+    MockBankAccountService.setPlaidClient.calls.reset();
+
+    // Default
+    component.bootstrapPlaid('');
+
+    // Previously loaded client
+    MockBankAccountService.getPlaidClient.and.returnValue(of(true));
+    component.bootstrapPlaid('');
+
+    // Ensure only one client has been loaded
+    expect(MockBankAccountService.setPlaidClient).toHaveBeenCalledOnceWith(
+      true
+    );
+  });
+
+  it('should handle errors when bootstrapping plaid', () => {
+    MockBankAccountService.getPlaidClient.and.returnValue(error$);
+
+    component.bootstrapPlaid('');
+
+    expect(MockErrorService.handleError).toHaveBeenCalled();
+    expect(MockEventService.handleEvent).toHaveBeenCalled();
+  });
+
+  it('should handle plaidOnSuccess()', () => {
```

Comment:
It's best practice to test one thing, i.e., success, invalid asset, or undefined asset, in each test as oppose to all in the same test.

Also, do we do any test to ensure we only got one account back?

Reply:
I've added a check for a single account.